addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3068,6 +3120,50 @@ export class OdooEditor extends EventTarget {
             this._positionToolbar();
         }
     }
+
+    /**
+     * Finds the font size to display for the current selection.
+     *
+     * @private
+     * @param {Selection} sel The current selection.
+     * @returns {Float} The font size to display.
+     */
+    _getFontSizeState(sel) {
+        const closestStartContainerEl = closestElement(sel.getRangeAt(0).startContainer, "*");
+        const closestFontSizedEl = closestStartContainerEl.closest(
+        `[style*='font-size'],${[...weUtils.FONT_SIZE_CLASSES.map(className => `.${className}`)]}` +
+        `,${[...weUtils.FONT_SIZE_CLASSES]}`
+        );
+        if (closestFontSizedEl) {
+            const customFontSize = closestFontSizedEl.style.fontSize;
+            if (customFontSize) {
+                if (customFontSize.includes("rem")) {
+                    const remValue = parseFloat(customFontSize);
+                    return weUtils.convertNumericToUnit(remValue, "rem", "px");
+                }
+                return parseFloat(/\d+/.exec(customFontSize).pop());
+            }
+            // It's a class font size or a hN tag. We don't return the computed
+            // font size because it can be different from the one displayed in
+            // the toolbar because it's responsive.
+            let fontSizeClass = weUtils.FONT_SIZE_CLASSES.find(
+                className => closestFontSizedEl.classList.contains(className));
+            const tagName = closestFontSizedEl.tagName.toLowerCase();
+            if (!fontSizeClass && weUtils.FONT_SIZE_CLASSES.includes(tagName)) {
+                fontSizeClass = tagName;
+            }
+            return parseFloat(this.toolbar.querySelector(
```

Comment:
I think it's a bit sad that this function looks like an util but ends up being a method of the editor just because we need `this` here.

Reply:
If you prefer, I can probably move it to a utility function and add a HTML element as param ?